### ISSUE TITLE
Multiple regions support

### ISSUE THREAD

**Issue by 0xtf** [wata727/packer-post-processor-amazon-ami-management#7]
Hi,

Is it possible to have the post processor utilize **{{user `destination_regions`}}**, for example:

>  "post-processors":[{
>      "type": "amazon-ami-management",
>      "region": "{{user `destination_regions`}}",
>      "identifier": "misp-build",
>      "keep_releases": "1"
>     }
> ]

Since I don't specify regions in environment variables, I'd still like to avoid specifying regions in two locations of the packer template.

**Comment by 0xtf** [comment]
The plugin works fine when using user variables. For example:

```
{
  "variables": {
    "aws_access_key": "{{env `AWS_ACCESS_KEY_ID`}}",
    "aws_secret_key": "{{env `AWS_SECRET_ACCESS_KEY`}}",
    "destination_regions": "eu-west-1,eu-west-2,eu-west-3"
 }
```

And using them in the plugin:

```
   "post-processors":[{
     "type": "amazon-ami-management",
     "region": "{{user `destination_regions`}}",
     "identifier": "misp-build",
     "keep_releases": "1"
    }
  ]
```

Closing  the issue.

**Comment by 0xtf** [comment]
Apparently I was too soon to say it worked. Using that configuration resulted in the following error:

```
1521471887,,ui,error,Build 'amazon-ebs' errored: 1 error(s) occurred:\n\n* Post-processor failed: RequestError: send request failed\ncaused by: Post https://ec2.eu-west-1%!(PACKER_COMMA)eu-west-2%!(PACKER_COMMA)eu-west-3.amazonaws.com/: dial tcp: lookup ec2.eu-west-1%!(PACKER_COMMA)eu-west-2%!(PACKER_COMMA)eu-west-3.amazonaws.com: no such host
1521471887,,error-count,1
1521471887,,ui,error,\n==> Some builds didn't complete successfully and had errors:
1521471887,amazon-ebs,error,1 error(s) occurred:\n\n* Post-processor failed: RequestError: send request failed\ncaused by: Post https://ec2.eu-west-1%!(PACKER_COMMA)eu-west-2%!(PACKER_COMMA)eu-west-3.amazonaws.com/: dial tcp: lookup ec2.eu-west-1%!(PACKER_COMMA)eu-west-2%!(PACKER_COMMA)eu-west-3.amazonaws.com: no such host
1521471887,,ui,error,--> amazon-ebs: 1 error(s) occurred:\n\n* Post-processor failed: RequestError: send request failed\ncaused by: Post https://ec2.eu-west-1%!(PACKER_COMMA)eu-west-2%!(PACKER_COMMA)eu-west-3.amazonaws.com/: dial tcp: lookup ec2.eu-west-1%!(PACKER_COMMA)eu-west-2%!(PACKER_COMMA)eu-west-3.amazonaws.com: no such host

```

**Comment by wata727** [comment]
Hi @0xtf,

You can use user variables, but this plugin doesn't accept multiple regions. `region` attribute type is string.

**Comment by 0xtf** [comment]
Hi @wata727,

Thank you for the reply. I noticed that user variables worked, but not with multi-region. 

Any plans on adding this feature? Would be amazing to have the plugin doing everything instead of having to rely on external aws-cli scripts.

**Comment by wata727** [comment]
Yes. I'm working on this feature 😉

**Comment by 0xtf** [comment]
Wow! That’s amazing! Thank you very much! Looking forward to it. If you need testing just let me know. Thanks!!

**Comment by wata727** [comment]
@0xtf I have been released v0.3.0 including this feature. Please check it.

**Comment by 0xtf** [comment]
Just finished testing! AWESOME! Working like a charm! Thank you so much for your work!
